module2/misc/deploying_with_render.md

```diff
@@ -0,0 +1,60 @@
+
+<p>Turing is not associated or affiliated with Render.com in any way; this page was created as a reference for students to deploy their applications for the first time. 
+</p>
+</section>
+
+## When should I deploy?
+- Deploy to Render (or more generally, the _production_ environment) when your application has reached a new apex in feature availability - in other words, when your `main` branch has been updated to where you are confident that you want it to be available on the internet. 
+- When deploying, Render will take ___your remote copy___, you'll be able to select which branch (we recommend `main`), and deploy it to your server. So, follow good git practice and only merge to `main` once your team has decided the code quality is good enough to deploy a new version of the app. 
+
+## Getting Started
+1. Create an account on Render, if you haven't already. (You can use GitHub to create an account to make your sign-in even easier.)
+
+2. Locally in the repository you'll be deploying, update the secret key before deploying: 
+   1. Delete the existing `config/credentials.yml.enc` file. 
+   2. In Terminal in this repository, run `rails credentials:edit`.
+         - You may need to specify `EDITOR="code" bin/rails credentials:edit` for it to work, otherwise it may give you errors. When successful, it will generate 2 files called `credentials.yml.enc` and `master.key`. The `master.key` file will be ignored by git - this is normal, you do not want to share this key with other teammates. 
+   3. Commit and push the change that added the new `credentials.yml.enc` file to your remote. 
+
+## Deploying an Existing Rails App
+
+### Create DB Service
+1. After you sign in, it will take you to your Dashboard. 
+2. Click the button to **Create a New PostgreSQL**. 
+   - Give the PostgreSQL service a name (recommend to use the same name as your repo for sanity)
+   - You can keep the **Database** and **User** fields blank, they will be randomly generated.
+   - **Change the PostgreSQL Version to 14.**
+   - Keep the Instance Type set to **Free**, and then click `Create Database`. 
+3. Once the database is created, click the `Connect` dropdown on the top-right, and copy the **Internal Database URL**. You'll use this link to connect to the production DB after you create the web service.
+
+### Create Web Service
+1. Click the button to Create a New __Web Service__. If you've connected your GitHub account, it will display some of your repositories that you can connect to for this web service. Click `Connect` next to the repo you'd like to deploy. 
+1.  Give the service a name (anything, you can use the same name as the repo)
+1.  In the **Build Command** step, add `rails db:{migrate,seed}` to the end of the string. 
```

Comment:
Will this cause duplicate data to be seeded? I'm not sure if this is the right place to do so, but maybe we could recommend that their `seeds.rb` includes any applicable `Record.destroy_all`s

Reply:
Agreed - it does create duplicates whenever a new build succeeds. I think having them put `destroy_all` in their seeds is better than overriding the `DISABLE_DATABASE_ENVIRONMENT_CHECK=1` in the prod environment too. I'll add this alert to the docs.